### PR TITLE
Making DateTime data zero-copy

### PR DESCRIPTION
**Pull request by robertbastian**

### CODE REVIEW

**Review by Manishearth — APPROVED** [review]
lgtm, probably need a data regen for zeromap

**Comment by robertbastian** [comment]
I did, which is why `testdata.postcard` is +81kb, which I find a bit strange.

**Comment by Manishearth** [comment]
the zeromaps are a _little bit_ less efficient than litemaps, serialization-wise, but i'm surprised it adds up to that much

File: components/datetime/src/provider/calendar/symbols.rs [hunk 1]
```diff
@@ -56,24 +59,28 @@ macro_rules! symbols {
         ($name: ident { $element: ident: Option<$ty: ty>, $($tokens: tt)+ } -> ($($members:tt)*)) => {
             symbols!($name { $($tokens)* } -> (
                 $($members)*
+                #[cfg_attr(feature = "provider_serde", serde(borrow))]
                 pub $element: Option<$ty>,
```
Comment (sffc) [inline components/datetime/src/provider/calendar/symbols.rs:63]: Issue: serde(borrow) does not work through options.

**Review by sffc — COMMENTED** [review]
Post-submit review (please follow the CONTRIBUTING.md please and hold off on merging until there are no more pending reviews)

**Comment by sffc** [comment]
I would also like to know where the 81 KB comes from.  I am uncomfortable with this being in 0.5 and this should not have been merged yet.

File: components/datetime/src/provider/calendar/symbols.rs (hunk 1 above)
Comment (sffc) [reply components/datetime/src/provider/calendar/symbols.rs:63]: Please add tests to verify that the data being deserialized is actually being borrowed.  `#[serde(borrow)]` is weird and it sometimes works and sometimes doesn't work.

Comment (Manishearth) [reply components/datetime/src/provider/calendar/symbols.rs:63]: Ugh i was looking for `Option` but didn't notice this one since it's in a macro

Comment (Manishearth) [reply components/datetime/src/provider/calendar/symbols.rs:63]: We can probably get a clippy lint that warns about serde(borrow) on option fields

**Pull request merged**